glib/src/gstring.rs

```diff
@@ -355,11 +530,18 @@ impl<'a> ToGlibPtr<'a, *mut c_char> for GStr {
 /// `GString` is to <code>&[GStr]</code> as [`String`] is to <code>&[str]</code>: the former in
 /// each pair are owned strings; the latter are borrowed references.
 ///
-/// This type is very similar to [`std::ffi::CString`], but with one added constraint: the string
-/// must also be valid UTF-8.
+/// This type is similar to [`std::ffi::CString`], but with some special behavior. When debug
+/// assertions are enabled, <code>[From]&lt;[String]></code> will panic if there are interior
+/// nul-bytes. In production builds, no checks will be made for interior nul-bytes, and strings
+/// that contain interior nul-bytes will simply end at first nul-byte when converting to a C
+/// string.
+///
+/// The constructors beginning with `from_utf8` `and `from_string` can also be used to further
+/// control how interior nul-bytes are handled.
+#[repr(transparent)]
```

Comment:
Why? This only really makes sense if you need guarantees about the representation, which we don't really here?

Reply:
I think this should at least give us some assurance it remains the same size as `String`?

Reply:
`String` is also not `#[repr(transparent)]` though :) I think if that's your goal then the only thing that you can really do here would be to add a test that asserts this, and if that ever fails we need to do some thinking.

Reply:
Removed the `repr` and added a test